docs/contribution/release.md

```diff
@@ -0,0 +1,78 @@
+# how release works
+
+Prerequisite:
+
+* docker installed
+* How drone works
+* How [goreleaser](https://goreleaser.com/intro/) works
+* multi arch TODO, don't know yet
+
+Briefly at least.
+
+## Continuous Delivery
+
+We use drone as continuous integration and continuous delivery pipeline.
+Integration means: running tests, checking code quality.
+Delivery means: make a release when a new tag get pushed.
+
+## goreleaser
+
+GoReleaser is a popular tool to release Go applications and library.
+
+I use it a lot because it is very flexible and it supports a lot of features, we
+need only a subset of them:
+
+* Multi arch build. It uses Go cross compilation feature to build for Darwin and
+  Linux (arm and amd)
+* Docker to build docker images for Linux container arm and amd.
+* Changelog to generate a changelog that will be pushed in the GitHub release
+  page with the list of PR part of the release itself.
+* Artifact (binaries) will be pushed as part of GitHub release page as well as
+  the other YAML file required by cluster-api like: metadata.yaml,
+  infrastructure-provider.yaml and so on.
+
+## Directories
+
+The release life cycle touches two directories:
+
+1. `./dist` it is a temporary directory used by gorleaser to bundle all the
+   required files and binaries as long as the generated changelog and so on. It
+   does not need to be checked in git and that's why it is in the gitignore.
+2. `./out` is ignored as well and it gets generated via `make release`. It is a
+   bundle that contains generated Kubernetes manifest required by the
+   cluster-api such as: metadata.yaml, infrastructure-provider.yaml,
+   cluster-template.yaml and so on.
+
+## Local workflow
+
+GoReleaser can be used locally, as well to test a release for example running:
+
+```
+$ goreleaser release --rm-dir --skip-publish
+```
+
+You can run this command if your git HEAD has a tag. You can make a temporary
+one:
+
+```
+git tag v0.10.0
+```
+
+Or you can decided to run GoRelease in `--snapshot` mode to avoid a fake tag
+(but changelog won't work):
+
+```
+$ goreleaser release --rm-dir --snapshot --skip-publish
+```
+
+*Removing --skip-publish goreleaser will attempt a push to GitHub and to your
+docker image repository (docker.io). In this way you can cut a release from your
+laptop if needed. But it is not a good idea and we use drone for that*
```

Comment:
Yeah, good point. Another reason for wrapping it in a make target, so that by default it doesn't publish, and you have to do something like `make release-publish CONFIRM=true` to get it.

Reply:
Might want to say that write access to the docker registry normally is blocked, but don't try it unless you really know what you are doing.